_integration/testsuite/make-kind-cluster.sh

```diff
@@ -0,0 +1,71 @@
+#! /usr/bin/env bash
+
+# Copyright 2020 VMware, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.  You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+set -o pipefail
+set -o errexit
+set -o nounset
+
+readonly KIND=${KIND:-kind}
+readonly KUBECTL=${KUBECTL:-kubectl}
+
+readonly CLUSTERNAME=${CLUSTER:-contour-integration}
+readonly WAITTIME=${WAITTIME:-5m}
+
+readonly HERE=$(cd $(dirname $0) && pwd)
+readonly REPO=$(cd ${HERE}/../.. && pwd)
+
+kind::cluster::exists() {
+    ${KIND} get clusters | grep -q "$1"
+}
+
+kind::cluster::create() {
+    ${KIND} create cluster \
+        --name "${CLUSTERNAME}" \
+        --wait "${WAITTIME}" \
+        --config "${REPO}/examples/kind/kind-expose-port.yaml"
+}
+
+kind::cluster::load() {
+    ${KIND} load docker-image \
+        --name "${CLUSTERNAME}" \
+        "$@"
+}
+
+if kind::cluster::exists "$CLUSTERNAME" ; then
+    echo "cluster $CLUSTERNAME already exists"
+    exit 2
+fi
+
+# Build the current version of Contour.
+make -C ${REPO} container IMAGE=docker.io/projectcontour/contour VERSION="v$$"
```

Comment:
To me, building a kind cluster and then building an image should be separate things. What if I want to use an existing cluster and skip that step?

Reply:
This script is specifically for the case where you want to test what you have in your repository. If you have an existing cluster, you don't need it.